pennylane/optimize/gradient_descent.py

```diff
@@ -47,86 +48,116 @@ def update_stepsize(self, stepsize):
         """
         self._stepsize = stepsize
 
-    def step_and_cost(self, objective_fn, x, grad_fn=None):
+    def step_and_cost(self, objective_fn, *args, grad_fn=None, **kwargs):
         """Update x with one step of the optimizer and return the corresponding objective
         function value prior to the step.
 
         Args:
             objective_fn (function): the objective function for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            *args : Variable length argument list for objective function
             grad_fn (function): Optional gradient function of the
                 objective function with respect to the variables ``x``.
                 If ``None``, the gradient function is computed automatically.
+                Must match shape of autograd derivative.
```

Comment:
Might be nicer to clarify a bit better what shape means in this case. E.g. "will always be a tuple containing ...".